src/command/list/mod.rs

```diff
@@ -141,7 +141,7 @@ impl Package {
     fn source(name: &str, project: Option<&Project>) -> Source {
         match project {
             Some(project) if project.has_direct_dependency(name) => {
-                Source::Project(project.package_file())
+                Source::Project(project.manifest_file().to_owned())
```

Comment:
I did a quick test, and it looks like it's definitely possible to use references throughout, however on balance I'm not sure if it's worth it. The upside is reducing a few file path copies on a cold path, while the downside is that the code becomes significantly more cluttered with explicit lifetime specifiers everywhere.

Perhaps if we decide to revisit the `volta list` code overall it would be good to investigate, but for the moment I think the cloning is worth it for the readability gains.

Reply:
Seems reasonable! I think that roughly matches my original reasoning in this space, too!